src/App/Services/Tamara_Client.php

```diff
@@ -54,6 +75,467 @@ public function reinit_tamara_client( $api_token, $api_url = 'https://api.tamara
 		static::$instance->api_client = $client;
 	}
 
+	/**
+	 * Create Tamara Checkout session, if failed, put errors to `wc_notices`
+	 *
+	 * @param $wc_order_id
+	 *
+	 * @return array|bool
+	 * @throws \Exception
+	 */
+	public function tamara_checkout_session( $wc_order_id ) {
+		$wc_order = wc_get_order( $wc_order_id );
+		$instalment_period = 3;
+		$checkout_payment_type = 'PAY_BY_INSTALMENTS';
+		try {
+			$checkout_response = $this->create_tamara_checkout_session(
+				$wc_order,
+				$checkout_payment_type,
+				$instalment_period
+			);
+		} catch ( Exception $tamaraCheckoutException ) {
+			$errorMessage = $this->tamara_checkout_wp_plugin->_t( 'Tamara Service unavailable! Please try again later.' );
+			if ( function_exists( 'wc_add_notice' ) ) {
+				wc_add_notice( $errorMessage, 'error' );
+			}
+		}
+
+		if ( isset( $checkout_response ) && $checkout_response->isSuccess() ) {
+			$tamara_checkout_url = $checkout_response->getCheckoutResponse()->getCheckoutUrl();
+			$tamara_checkout_session_id = $checkout_response->getCheckoutResponse()->getCheckoutId();
+
+			update_post_meta( $wc_order_id, 'tamara_checkout_session_id', $tamara_checkout_session_id );
+			update_post_meta( $wc_order_id, '_tamara_checkout_session_id', $tamara_checkout_session_id );
+			update_post_meta( $wc_order_id, 'tamara_checkout_url', $tamara_checkout_url );
+			update_post_meta( $wc_order_id, '_tamara_checkout_url', $tamara_checkout_url );
+			update_post_meta( $wc_order_id, 'tamara_payment_type', $checkout_payment_type );
+			update_post_meta( $wc_order_id, '_tamara_payment_type', $checkout_payment_type );
+			if ( $checkout_payment_type === 'PAY_BY_INSTALMENTS' && ! empty( $instalment_period ) ) {
+				update_post_meta( $wc_order_id, 'tamara_payment_type_instalment', $instalment_period );
+				update_post_meta( $wc_order_id, '_tamara_payment_type_instalment', $instalment_period );
+			}
+
+			return [
+				'result' => 'success',
+				'redirect' => $tamara_checkout_url,
+				'tamara_checkout_url' => $tamara_checkout_url,
+				'tamara_checkout_session_id' => $tamara_checkout_session_id,
+			];
+		}
+		// If this is the failed process, return false instead of ['result' => 'success']
+		return false;
+	}
+
+	/**
+	 * Create Tamara Checkout Session Request
+	 *
+	 * @param  WC_Order  $wc_order
+	 *
+	 * @param $payment_type
+	 * @param $instalment_period
+	 *
+	 * @return CreateCheckoutResponse
+	 * @throws Exception
+	 */
+	public function create_tamara_checkout_session( WC_Order $wc_order, $payment_type, $instalment_period ): CreateCheckoutResponse {
```

Comment:
we may rename this to `build_checkout_session_request()` as we are in the Tamara_Client so we don't need to refer to tamara again